index.js

```diff
@@ -151,6 +151,10 @@ module.exports = function (newrelic, opts) {
 	}
 
 	return async function koaNewrelic(ctx, next) {
+		if (opts.preSetTransaction) {
+			setTransactionName(ctx.method, ctx.path);
+		}
+
 		await next();
 
 		if (ctx._matchedRoute) {
```

Comment:
https://github.com/alexmingoia/koa-router/tree/master#routergetputpostpatchdeletedel--router
`When a route is matched, its path is available at ctx._matchedRoute and if named, the name is available at ctx._matchedRouteName`